### PR TITLE
Transfer in Geometry.x changes from RT branch

### PR DESCRIPTION
**Pull request by bgbsww**
Existing methods appear to be tested by the existing python suite.  This passes all python and c++ tests.

### CODE REVIEW

File: src/Mod/Part/App/Geometry.cpp [hunk 1]
```diff
@@ -61,14 +63,20 @@
 # include <Geom_SurfaceOfRevolution.hxx>
 # include <Geom_ToroidalSurface.hxx>
 # include <Geom_TrimmedCurve.hxx>
+# if OCC_VERSION_HEX < 0x070600
+#   include <GeomAdaptor_HSurface.hxx>
+#   include <GeomAdaptor_HCurve.hxx>
```
Comment (chennes) [inline src/Mod/Part/App/Geometry.cpp:68]: This one already appears below

File: src/Mod/Part/App/Geometry.h [hunk 2]
```diff
@@ -153,7 +158,7 @@ class PartExport GeomPoint : public Geometry
     void Restore(Base::XMLReader &/*reader*/) override;
     // Base implementer ----------------------------
     PyObject *getPyObject() override;
-
+    bool isSame(const Geometry &other, double tol, double atol) const override;
```
Comment (chennes) [inline src/Mod/Part/App/Geometry.h:161]: Can you add a docstring about the two doubles?

File: src/Mod/Part/App/Geometry.h [hunk 3]
```diff
@@ -207,8 +218,11 @@ class PartExport GeomCurve : public Geometry
 
     Base::Vector3d value(double u) const;
 
+    GeomLine *toLine(bool clone=true) const;
+    GeomLineSegment *toLineSegment(bool clone=true) const;
```
Comment (chennes) [inline src/Mod/Part/App/Geometry.h:222]: You'll no doubt be shocked to hear that I don't like leaving this as a boolean.

Comment (bgbsww) [reply src/Mod/Part/App/Geometry.h:222]: Laugh!  Totally agree with this one.  The lack of meaning clarity is brutal.

**Pull request merged**